src/components/NumberTicker.tsx

```diff
@@ -43,23 +40,22 @@ function Tick({ startValue, endValue, textHeight, textStyles, animationDuration
   }).start()
 
   return (
-    <Animated.View style={[transformStyle]}>
+    <Animated.View style={transformStyle}>
       {numberRange.map((number, index) => {
-        return (
-          <TickText key={index} textHeight={textHeight} textStyles={textStyles} value={number} />
-        )
+        return <TickText key={index} textStyles={textStyles} value={number} />
       })}
     </Animated.View>
   )
 }
 
 export default function NumberTicker({
   finalValue,
-  textStyles,
-  textHeight,
-  animationDuration,
+  typeScaleName = 'displaySmall',
+  animationDuration = 1300,
   testID,
 }: Props) {
+  const textStyles = typeScale[typeScaleName]
```

Comment:
having a prop with a style name to looks too smart to me: why not just provide a `style` as it is? in that case we still can extract a `lineHeight` from it, right?

also, what if the style doesn't have a `lineHeight` property? perhaps we should have a reasonable default? something like 1.2

Reply:
2 reasons: 
1. once we've described a prop as a StyleProp in typescript, you can't reference the style properties easily anymore
2. exactly what you said - what if the style doesn't have a lineHeight? so i think it is better to have a prop that enforces us to use `typeScale` and not pass in unexpected combinations of styles

Reply:
but this implies that `typeScale` must have the `lineHeight` prop. if we remove that prop from it, this component will break. looks like we are requiring the `typeScale` to have a specific shape from now on.

what is we use [StyleSheet.flatten()](https://reactnative.dev/docs/stylesheet#flatten)?

then we could extract the line height somehow like this:
```typescript
const textHeight = StyleSheet.flatten(textStyle).lineHeight ?? DEFAULT_TEXT_HEIGHT
```

the main benefit I see the consumer can provide any style and it still works.

Reply:
i'm not sure that we can or try to strive for "the consumer can provide any style and it still works". if a consumer applies a style that has no lineHeight + a fontSize that is bigger than DEFAULT_TEXT_HEIGHT, then this still breaks. 

if i make a style prop that requires fontSize and lineHight to be both defined, the consumer can _still_ add a vertical padding or margin that can screw up the styles.

i think this is a byproduct of trying to make something cater for too many scenario (i.e. trying to allow all styles to come through means we need to think about all combinations that could not work) and it's unnecessary complexity to maintain. i prefer to keep the degree of customisability to a minimum, which is similar about design system components. wdyt?

Reply:
yes, i agree! 
there are still many ways for consumers to screw that up.
i also like the idea of sticking with a design system discouraging arbitrary customization. 

apologies for introducing the confusion -- i was mistakenly treating `lineHeight` numeric values as _ratios_ similar to CSS ones, while they are absolute values in RN 😖

Reply:
i did a quick check to make sure that there are typescript errors when `lineHeight` is removed on any object in `typeScale`, so we reduce the risk of accidentally breaking this. (of course styles can be _added_ to `typesScale` to break this component though 😅)